src/com/utils/SudokuGenerator.java

```diff
@@ -0,0 +1,189 @@
+package com.utils;
+
+import com.algorithm.Algorithm;
+import com.algorithm.Backtracking;
+import com.sudoku.Cell;
+import com.sudoku.SudokuBoard;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class that represent the Sudoku Generator class
+ */
+public class SudokuGenerator {
+
+    private static SudokuBoard board = new SudokuBoard();
+
+    public enum Level {
+        EASY,
+        MEDIUM,
+        HARD
+    }
+
+    public static Map<Level, Integer> levels = new HashMap<Level, Integer>();
+
+    static {
+        levels.put(Level.EASY, 3);   // max 3 empty cells per sub grid
+        levels.put(Level.MEDIUM, 5); // max 5 empty cells per sub grid
+        levels.put(Level.HARD, 7);   // max 7 empty cells per sub grid
+    }
+
+    public static void generateEasy() {
```

Comment:
I dont think have these methods will help at all, instead that means if the levels change, that will affect the code, i will suggest to remove them and let only the generic one that receives the level